server/bin/pbench-server-prep-shim-002.py

```diff
@@ -110,147 +79,41 @@ def process_tb(config, logger, receive_dir, qdir_md5, duplicates, errors):
         os.path.join(receive_dir, "**", "*.tar.xz.md5"), recursive=True
     )
 
-    archive = config.ARCHIVE
     logger.info("{}", config.TS)
     list_check.sort()
     nstatus = ""
 
-    ntotal = ntbs = nerrs = nquarantined = ndups = 0
+    ntotal = ntbs = nerrs = 0
+
+    # FIXME: keep a generated token which has longer duration
+
+    token = config.get("put-server", "token")
```

Comment:
Again, I don't like the idea of a `put-server` section; that feels way too specific, even though in this particular case the stored token __will__ only be used for the prep-shim PUT function. I think I'd still suggest putting this under the `pbench-server` section along with the base URI, and maybe rename it `put-token` (as eventually we'll want this to be restricted to the PUT role).

Reply:
As I said in my previous comment, I'm expecting the shim to be a "one off" deployment -- it's critical, but it's unique, limited in scope, and limited in time.  So, I would vote for minimizing it's impact of the structures of the Pbench server, itself.

So, I would vote against adding new config files, and I would vote against adding new fields to existing sections of existing config files.  I think it would be best if the stuff which is specific to the shim appeared under its own section of the existing server (?) config file -- that way, when we're done with the shim (or when we deploy elsewhere without shim) there is no trace.

I don't think we need a lot of "documentation" or other lingering support (like config templates) for the shim:  we need the shim during the transition period from 0.69 to 0.72 or so, we can deploy it manually, adding the config stuff "by hand", and, once we get to 0.72, the shim and everything that it depends on should disappear.  So, it would be nice if we planned for that, now.

Reply:
I have reverted the changes to config files and made changes to keys and values. For now, I have left the put-shim.cfg with some new changes. We can remove that before merging.